tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -106,34 +108,40 @@ public Span? RootSpan
                 spans.Array![spans.Offset].Context.TraceContext :
                 null;
 
-        internal void AddWafSecurityEvents(IReadOnlyCollection<object> events)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal AppSecRequestContext GetRequestContext()
         {
             if (Volatile.Read(ref _appSecRequestContext) is null)
             {
                 Interlocked.CompareExchange(ref _appSecRequestContext, new(), null);
             }
 
-            _appSecRequestContext!.AddWafSecurityEvents(events);
+            return _appSecRequestContext!;
         }
 
-        internal void AddStackTraceElement(Dictionary<string, object> stack, int maxStackTraces)
+        internal void AddWafSecurityEvents(IReadOnlyCollection<object> events)
         {
-            if (Volatile.Read(ref _appSecRequestContext) is null)
-            {
-                Interlocked.CompareExchange(ref _appSecRequestContext, new(), null);
-            }
+            GetRequestContext().AddWafSecurityEvents(events);
+        }
 
-            _appSecRequestContext!.AddRaspStackTrace(stack, maxStackTraces);
+        internal void AddRaspStackTraceElement(Dictionary<string, object> stack, int maxStackTraces)
+        {
+            GetRequestContext().AddRaspStackTrace(stack, maxStackTraces);
         }
 
-        internal void AddRaspSpanMetrics(ulong duration, ulong durationWithBindings, bool timeout)
+        internal string? GetNextVulnerabilityStackTraceId()
         {
-            if (Volatile.Read(ref _appSecRequestContext) is null)
-            {
-                Interlocked.CompareExchange(ref _appSecRequestContext, new(), null);
-            }
+            return _iastRequestContext?.GetNextVulnerabilityStackId();
+        }
```

Comment:
Maybe we should just expose `_iastRequestContext` in the same way you have for `_appSecRequestContext` ?

Reply:
As discussed, we'll consider that in a separate PR